### PR TITLE
Started work on skeleton quest generation stuff

### PR DESCRIPTION
**Pull request by andrewgotow**

### CODE REVIEW

File: src/main/scala/ayai/quests/FetchObjective.scala [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+class FetchObjective(name: String, target: String, recipient: String) extends Objective(name) {
+  override def asJson(): JObject = {
+    ("name" -> name) ~
+      ("target" -> target) ~
+      ("recipient" -> recipient)
+  }
+
+  override def isComplete(): Boolean = {
+    // do something fancy here to check if the target has been delivered.
+    return false
```
Comment (chrisbenincasa) [inline src/main/scala/ayai/quests/FetchObjective.scala]: [nit] `return` not needed if the statement is the last in a function

File: src/main/scala/ayai/quests/KillObjective.scala [hunk 2]
```diff
@@ -9,4 +9,8 @@ class KillObjective(name: String, totalCompleted: Int, totalNeeded:Int) extends
 		("totalCompleted" -> totalCompleted) ~
 		("totalNeeded" -> totalNeeded)
 	}
+
+  override def isComplete(): Boolean = {
```
Comment (chrisbenincasa) [inline src/main/scala/ayai/quests/KillObjective.scala]: [nit] generally, only functions which have [side effects](http://en.wikipedia.org/wiki/Side_effect_%28computer_science%29) have empty-parens

File: src/main/scala/ayai/components/Quest.scala [hunk 3]
```diff
@@ -1,14 +1,14 @@
 package ayai.components
 
 import crane.Component
-import ayai.quests.{KillObjective}
+import ayai.quests.{Objective}
```
Comment (chrisbenincasa) [inline src/main/scala/ayai/components/Quest.scala]: [supernit] single import doesn't need brackets.

**Comment by chrisbenincasa** [comment]
Looks like this needs a rebase. Code looks good, just a coupe of nits.

**Comment by andrewgotow** [comment]
Ok! Forgive my ignorance, but I think I've got the pull-request thing pretty much figured out. Don't merge this request yet though, I've discovered some issues with loading quests from the JSON files that I still need to work out.

**Comment by chrisbenincasa** [comment]
No worries! That last commit has a ton of changes `+2003/-2024`, some of which have already been merged in I think (Perception stuff). Try this:

```
$> git remote -v
origin  https://github.com/ayai-ai/ayai-backend.git (fetch)
origin  https://github.com/ayai-ai/ayai-backend.git (push)
$> git fetch origin
$> git rebase -i origin/develop
```

(also, yours might be called something other than `origin`, but you get the idea :stuck_out_tongue: )

**Pull request merged**